lib/ansible/modules/cloud/amazon/elasticsearch.py

```diff
@@ -0,0 +1,421 @@
+        (('EBSOptions', 'VolumeType'), desired_config['ebs_volume_type']),
+        (('EBSOptions', 'VolumeSize'), desired_config['ebs_volume_size']),
+    ]
+
+    if desired_config['dedicated_master_enabled']:
+        config_to_check.extend(
+            [
+                (('ElasticsearchClusterConfig', 'DedicatedMasterType'),
+                    desired_config['dedicated_master_type']),
+                (('ElasticsearchClusterConfig', 'DedicatedMasterCount'),
+                    desired_config['dedicated_master_count']),
+            ]
+        )
+
+    for key_path, desired_value in config_to_check:
+        # Traverse key path to get existing value
+        existing_value = existing_config
+        for key in key_path:
+            existing_value = existing_value.get(key)
+
+        if existing_value != desired_value:
+            # Traverse key path in modifications dictionary to submit to AWS
+            modification = modifications_needed
+            for key in key_path[:-1]:
+                if key not in modification:
+                    modification[key] = {}
+
+                modification = modification[key]
+
+            modification[key_path[-1]] = desired_value
+
+    existing_access_json = json.loads(domain['DomainStatus']['AccessPolicies'])
+    supplied_access_json = json.loads(module.params['access_policies'])
+
+    if compare_policies(existing_access_json, supplied_access_json):
+        modifications_needed['AccessPolicies'] = module.params['access_policies']
+
+    return modifications_needed
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            domain={'required': True},
+            state={'required': True, 'choices': ['present', 'absent']},
+            version={'default': '5.1'},
+            instance_count={'type': 'int', 'default': 1},
+            instance_type={'required': True},
+            zone_awareness={'type': 'bool', 'default': True},
+            dedicated_master_enabled={
+                'type': 'bool', 'default': False
+            },
+            dedicated_master_type={},
+            dedicated_master_count={'type': 'int'},
+            ebs_volume_type={
+                'default': 'gp2', 'choices': ['standard', 'gp2', 'io1']
+            },
+            ebs_volume_size={'required': True, 'type': 'int'},
+            access_policies={'required': True},
+            wait={'type': 'bool'},
+            tags={'type': 'dict'},
+        )
+    )
+
+    module = AnsibleAWSModule(
+        argument_spec=argument_spec,
+    )
+
+    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(
+        module, boto3=True
+    )
+
+    client = boto3_conn(
+        module, conn_type='client', resource='es', region=region,
+        endpoint=ec2_url, **aws_connect_kwargs
+    )
+
+    if module.params['state'] not in ['present', 'absent']:
+        module.fail_json(msg='"state" must be "present" or "absent"')
```

Comment:
This statement is not required as it's handled by Ansible parameter checking... you've already specified the choices.